cmake/modules/BuildBoost.cmake

```diff
@@ -67,10 +67,10 @@ function(do_build_boost root_dir version)
 
   set(boost_with_libs)
   foreach(c ${Boost_BUILD_COMPONENTS})
-    if(c MATCHES "^python([0-9])\$")
```

Comment:
probably we would never see python10. and according to Guido, python4 might never come to the world in the foreseeable future. so i dropped this change.